pkg/server/gateway.go

```diff
@@ -80,6 +80,13 @@ func NewGateway(address string, refresh time.Duration, caFile, certificateFile,
 		return nil, err
 	}
 
+	ctx, cancel := context.WithTimeout(context.Background(), waitForBalancer)
+	defer cancel()
+	_, _, err = balancer.Get(ctx, grpc.BalancerGetOptions{BlockingWait: true})
```

Comment:
If we pass the context through then this doesn't need to create `context.WithTimeout` it can just cancel according to the context it's provided.